templates/ingress.yaml

```
@@ -11,10 +11,17 @@ metadata:
     chart: "{{ .Chart.Name }}-{{ .Chart.Version | replace "+" "_" }}"
 spec:
   rules:
-  -  http:
+  - http:
       paths:
       - path: /
         backend:
           serviceName: {{ template "fullname" . }}
           servicePort: {{ .Values.service.externalPort }}
+    {{- if .Values.ingress.hostName }}
+    host: {{ .Values.ingress.hostName | quote }}
+    {{- end -}}
+  {{- if .Values.ingress.tls }}
+  tls:
+{{ toYaml .Values.ingress.tls | indent 2 }}
```

```suggestion
  tls: {{- toYaml .Values.ingress.tls | nindent 4 }}
```
`nindent` is the same as `indent` except it prepends a newline. This, along with the `-`, should help avoid whitespace shenanigans.